include/mxnet/tensor_blob.h

```diff
@@ -184,7 +184,7 @@ class TBlob {
         mshadow::Shape1(shape_.Size()), stream);
   }
   /*! \brief return number of dimension of the tensor inside */
-  inline int ndim(void) const {
+  inline index_t ndim(void) const {
     return shape_.ndim();
```

Comment:
Other codes may rely on the return type of ndim() and we should check more carefully before making the change.